src/Controller/RegistrationController.php

```diff
@@ -0,0 +1,95 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Exception\NoAuthnrequestException;
+use App\PublicKeyCredentialCreationOptionsStore;
+use App\Repository\UserRepository;
+use Psr\Log\LoggerInterface;
+use Surfnet\GsspBundle\Service\RegistrationService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Annotation\Route;
+use Webauthn\Bundle\Service\PublicKeyCredentialCreationOptionsFactory;
+
+final class RegistrationController extends AbstractController
+{
+    private $registrationService;
+    private $userRepository;
+    private $publicKeyCredentialCreationOptionsFactory;
+    private $creationOptionsStore;
+    private $logger;
+
+    public function __construct(
+        RegistrationService $registrationService,
+        UserRepository $userRepository,
+        PublicKeyCredentialCreationOptionsFactory $publicKeyCredentialCreationOptionsFactory,
+        PublicKeyCredentialCreationOptionsStore $creationOptionsStore,
+        LoggerInterface $logger
+    ) {
+        $this->registrationService = $registrationService;
+        $this->userRepository = $userRepository;
+        $this->publicKeyCredentialCreationOptionsFactory = $publicKeyCredentialCreationOptionsFactory;
+        $this->creationOptionsStore = $creationOptionsStore;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Replace this example code with whatever you need.
+     *
+     * See @see RegistrationService for a more clean example.
+     *
+     * @Route("/registration", name="app_identity_registration")
+     */
+    public function __invoke()
+    {
+        $this->logger->info('Verifying if there is a pending registration from SP');
+
+        if (!$this->registrationService->registrationRequired()) {
+            $this->logger->warning('Registration is not required');
+            throw new NoAuthnrequestException();
+        }
+
+        $this->logger->info('There is a pending registration');
+
+        $this->logger->info('Verifying if registration is finalized');
+
+        if ($this->registrationService->isRegistered()) {
+            $this->logger->info('Registration is finalized returning to service provider');
+            return $this->registrationService->replyToServiceProvider();
+        }
+
+        $this->logger->info('Registration is not finalized create public key credential creation options');
+
+        $userEntity = $this->userRepository->createUserEntity('SURFsecureID', 'SURFsecureID', null);
```

Comment:
Not sure we should use a SURFism here. This might need to be a configurable value!

Reply:
Moved to config, use nameId as username, for now display name is configurable with parameters.yml .